wx.ProgressDialog SetRange Problem

<!-- For bugs or other problems please provide the following details in addition to
     your issue report, if applicable. See also https://wxpython.org/pages/how-to-submit-issue/

     For issues about building on Linux, please read this page before reporting it here:
     https://wxpython.org/blog/2017-08-17-builds-for-linux-with-pip/
-->

**Operating system**:
windows7
**wxPython version**:
3.0.2
**Stock or custom build**:
**Python version**:
2.7
**Stock or custom build**:

**Description of the problem**:

I create a ProgressDialog:
dlg = wx.ProgressDialog("Progress dialog example",
"An informative message",
maximum = 100,
parent=self,
style = 0
| wx.PD_APP_MODAL
| wx.PD_CAN_ABORT
#| wx.PD_CAN_SKIP
#| wx.PD_ELAPSED_TIME
| wx.PD_ESTIMATED_TIME
| wx.PD_REMAINING_TIME
#| wx.PD_AUTO_HIDE
)
When I call dlg.SetRange method,It occur problem below:
Traceback (most recent call last):
File "D:\env\wxpython_test\thread_test_2.py", line 137, in onButton2
dlg.SetRange(120)
File "C:\Python27\lib\site-packages\wx-3.0-msw\wx_windows.py", line 3744, in SetRange
return windows.GenericProgressDialog_SetRange(*args, **kwargs)
wx._core.PyAssertionError: C++ assertion "m_gauge" failed at ....\src\generic\progdlgg.cpp(610) in wxGenericProgressDialog::SetRange(): dialog should
be fully created

But the dlg.Update and other method work ok.
this probem only occur on windows os system,It work fine on linux os system.
What's the probelm may be and How to avoid this ?

<!-- if possible please include a small runnable application that demonstrates the problem -->

```python
import time
import wx

########################################################################
class MyForm(wx.Frame):
 
    #----------------------------------------------------------------------
    def __init__(self):
        wx.Frame.__init__(self, None, wx.ID_ANY, "Tutorial")
 
        # Add a panel so it looks the correct on all platforms
        panel = wx.Panel(self, wx.ID_ANY)
        self.displayLbl = wx.StaticText(panel, label="Amount of time since thread started goes here")
        self.btn = btn = wx.Button(panel, label="Start Thread")
 
        btn.Bind(wx.EVT_BUTTON, self.onButton2)
 
        sizer = wx.BoxSizer(wx.VERTICAL)
        sizer.Add(self.displayLbl, 0, wx.ALL|wx.CENTER, 5)
        sizer.Add(btn, 0, wx.ALL|wx.CENTER, 5)
        panel.SetSizer(sizer)
 
        # create a pubsub receiver

    def onButton2(self,event):
        dlg = wx.ProgressDialog("Progress dialog example",
                               "An informative message",
                               maximum = 100,
                               parent=self,
                               style = 0
                                | wx.PD_APP_MODAL
                                | wx.PD_CAN_ABORT
                                #| wx.PD_CAN_SKIP
                                #| wx.PD_ELAPSED_TIME
                                | wx.PD_ESTIMATED_TIME
                                | wx.PD_REMAINING_TIME
                                #| wx.PD_AUTO_HIDE
                                )
        dlg.SetRange(120)
    #----------------------------------------------------------------------
    def updateDisplay(self, msg):
        """
        Receives data from thread and updates the display
        """
        t = msg
        if isinstance(t, int):
            self.displayLbl.SetLabel("Time since thread started: %s seconds" % t)
        else:
            self.displayLbl.SetLabel("%s" % t)
            self.btn.Enable()
 
#----------------------------------------------------------------------
# Run the program
if __name__ == "__main__":
    app = wx.PySimpleApp()
    frame = MyForm().Show()
    app.MainLoop()
```

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -106,7 +106,8 @@ Changes in this release include the following:
 
 * Fix displaying '&' in the label of wx.RadioBox on GTK. (#39)
 
-
+* Fix problems of the wrong C++ method being called in wx.ProgressDialog on MS
+  Windows. (#701)
 
 
 

--- a/etg/progdlg.py
+++ b/etg/progdlg.py
@@ -9,6 +9,7 @@
 
 import etgtools
 import etgtools.tweaker_tools as tools
+import copy
 
 PACKAGE   = "wx"
 MODULE    = "_core"
@@ -34,7 +35,7 @@ def run():
 
     module.addHeaderCode("#include <wx/progdlg.h>")
 
-    c = module.find('wxGenericProgressDialog')
+    c = gpd = module.find('wxGenericProgressDialog')
     assert isinstance(c, etgtools.ClassDef)
 
     tools.fixWindowClass(c)#, False)
@@ -46,6 +47,18 @@ def run():
 
     c = module.find('wxProgressDialog')
     tools.fixWindowClass(c)
+
+    # Copy methods from the generic to the native class. This is needed
+    # because none of the methods are declared in the interface files, and
+    # since on MSW some non-virtual methods are reimplemented in
+    # wxProgressDialogs they will not be called if SIP doesn't know about
+    # them. We'll copy all of them and let the C++ compiler sort things out.
+    for item in gpd:
+        if (isinstance(item, etgtools.MethodDef) and
+                not item.isCtor and
+                not item.isDtor):
+            c.addItem(copy.deepcopy(item))
+
 
     #-----------------------------------------------------------------
     tools.doCommonTweaks(module)

